arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -1629,7 +1630,38 @@ ErrorCode RocksDBMetaCollection::doLock(double timeout, AccessMode::Type mode) {
   auto timeout_us = std::chrono::duration_cast<std::chrono::microseconds>(
       std::chrono::duration<double>(timeout));
 
+  constexpr auto detachThreshold = std::chrono::microseconds(1000000);
+
   bool gotLock = false;
+  if (timeout_us > detachThreshold) {
+    if (mode == AccessMode::Type::WRITE) {
+      gotLock = _exclusiveLock.tryLockWriteFor(detachThreshold);
+    } else {
+      gotLock = _exclusiveLock.tryLockReadFor(detachThreshold);
+    }
```

Comment:
What it `timeout` is less than 1s? Then this could would still block for 1s. And if we it is greater than 1s but we do not get the lock hear, then the later tryLock does not account for the time spent here already, so in the worst case we block for `timeout + 1s`. Is that intentional?

Reply:
There was already a subtraction of `detachThreshold` from `timeout_us`, but IMHO it happened at the wrong location, so it could have made the timeout negative. I pushed a fixed for this now.